era/front.js

```diff
@@ -131,3 +150,508 @@ function setParamAndReload() {
+    el_targetSelect.innerHTML = "";
+    j = 0;
+    const sortedTargets = Object.keys(enemy.threat).map(
+      (k) => enemy.threat[k].target
+    );
+
+    sortedTargets.sort((a, b) => a.name.localeCompare(b.name));
+
+    for (let target of sortedTargets) {
+      let el_u = document.createElement("option");
+      el_u.value = enemyKey + ";" + target.key;
+      el_u.textContent = target.name + " - " + target.key;
+      el_targetSelect.appendChild(el_u);
+      if (target.key === prevSelection) el_targetSelect.selectedIndex = j;
+      j += 1;
+    }
+    selectTarget(config);
+  } catch (e) {
+    printError(e);
+  }
+}
+
+/**
+ * @param {import("./base.js").GameVersionConfig} config
+ * @param {string | undefined} [target]
+ */
+export function selectTarget(config, target) {
+  try {
+    let el = document.querySelector("#targetSelect");
+    let el_out = document.querySelector("#threatTableContainer");
+    el_out.innerHTML = "";
+    if (!target) {
+      let i = el.selectedIndex;
+      if (i === -1) return;
+      target = el.options[i].value;
+    } else {
+      for (let i = 0; i < el.options.length; ++i) {
+        if (el.options[i].value === target) {
+          el.selectedIndex = i;
+          break;
+        }
+      }
+    }
+    let [reportId, fightId, enemyId, targetId] = target.split(";");
+    const trace =
+      reports[reportId].fights[fightId].enemies[enemyId].threat[targetId];
+    tabulate(config, el_out, trace);
+  } catch (e) {
+    printError(e);
+  }
+}
+function enableInput(enable = true) {
+  let a = ["input", "button", "select"].map((s) =>
+    document.querySelectorAll(s)
+  );
+  for (let b of a) {
+    for (let el of b) {
+      el.disabled = !enable;
+    }
+  }
+}
+
+/**
+ * @param {Element | null} el_out
+ * @param {boolean} checked
+ * @param {string | null} text
+ * @param {{ (x: any): void; (x: any): void; (arg0: boolean): any; }} callback
+ */
+export function createCheckbox(el_out, checked, text, callback) {
+  let el_checkbox = document.createElement("input");
+  let el_label = document.createElement("label");
+  el_checkbox.type = "checkbox";
+  el_checkbox.className = "checkbox";
+  el_checkbox.checked = checked;
+  el_checkbox.onchange = () => callback(el_checkbox.checked);
+  el_label.textContent = text;
+  el_out.appendChild(el_checkbox);
+  el_out.appendChild(el_label);
+}
+/**
+ * @param {string} name
+ */
+function getParameterByName(name, url = window.location.href) {
+  name = name.replace(/[\[\]]/g, "\\$&");
+  var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+    results = regex.exec(url);
+  if (!results) return null;
+  if (!results[2]) return "";
+  return decodeURIComponent(results[2].replace(/\+/g, " "));
+}
+
+let plotData = [];
+
+/**
+ * @param {import("./base.js").GameVersionConfig} config
+ * @param {string} reportId
+ * @param {Fight} fight
+ * @param {NPC} enemy
+ */
+function plot(config, reportId, fight, enemy) {
```

Comment:
plot moved from `npc.plot(reportId, fight)` to here as `plot(reportid, fight, enemy)`

Keeps UI code separate so we can change the UI layer easier.